lib/services/files-api-service.js

```diff
@@ -0,0 +1,129 @@
+// Copyright 2016, EMC, Inc.
+
+'use strict';
+
+var di = require('di');
+var fs = require('fs');
+var path = require('path');
+var walking = require('walk');
+
+module.exports = filesApiServiceFactory;
+
+di.annotate(filesApiServiceFactory, new di.Provide('Http.Services.Api.StaticFiles'));
+di.annotate(filesApiServiceFactory,
+    new di.Inject(
+      'Constants',
+      'Promise',
+      'Services.Configuration',
+      'Services.Waterline'
+    )
+);
+
+
+function filesApiServiceFactory(
+    constants,
+    Promise,
+    configuration,
+    waterline
+) {
+    var walker;
+    var walk = Promise.promisifyAll(walking);
+    var skuPath = constants.HttpStatic.onhttpSkupack;
+    var staticPath = constants.HttpStatic.onhttpStaticHttp;
+    Promise.promisifyAll(fs);
+
+    function FilesApiService() {
+    }
+
+
+    FilesApiService.prototype.pairSkupackIds = function() {
+        return fs.readdirAsync(skuPath).then(function(list) {
+            return Promise.filter(list, function(item) {
+                return (fs.lstatAsync(path.join(skuPath, item)))
+                .then(function(stat) {
+                    return stat.isDirectory();
+                }); 
+            }).then(function(dirList) {
+                return Promise.reduce(dirList, function(result, item) {
+                    return waterline.skus.findOne({id : item})
+                    .then(function(sku) {
+                        result[item] = sku.name;
+                        return result;
+                    });
+                },{});
+            });
+        });
+    };
+
+
+    FilesApiService.prototype.walkDirectory = function(dir) {
+        return new Promise(function(resolve, reject) {
+            var walkResults = [];
+            walker = walk.walk(dir, {followLinks: false});
+
+            walker.on("file", function(root, fileStats, next) {
+                Promise.resolve().then(function() {
+                    var walkObj = {};
+                    if (root.indexOf(staticPath) > -1){
+                        root = root.replace(staticPath, '');
+                        walkObj.uri = path.join(root, fileStats.name);
+                        walkResults.push(walkObj);
+                    } else if ((root.indexOf(skuPath) > -1) && 
+                      (root.indexOf("static/common") > -1)){
```

Comment:
We support directories other than `static/common`, i.e. `static/centos` etc. Basically any directory or file within static is valid.